ui/apps/platform/src/hooks/useLayoutSpaceObserver.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * Hook to observe the cumulative amount of space a list of elements take up
+ * in a target parent element.
+ *
+ * @param root
+ *      The parent element container to observe. An `undefined` value will use `document.body`.
+ * @param observationTargets
+ *      An array of descendent elements of `root` whose space should be monitored.
+ * @param granularity
+ *      How frequently space updates should be reported. Higher numbers will result in more frequent
+ *      and more accurate updates but with a larger volume of function calls. Must be an integer >= 1.
+ * @return
+ *      The total `height` and `width` the target elements take up in the visible portion of the root element.
+ */
+export default function useLayoutSpaceObserver(
```

Comment:
This is used to track the visible space a list of elements occupy in a parent container. Although this was needed for the specific case of the Selector Results component, it was general enough that I moved it to the main `hooks/` directory. If we don't feel like this is something we are going to want to reuse I can co-locate it with the collections code.